scripts/ci/metrics.sh

```diff
@@ -133,3 +133,86 @@ _EO_UPDATE_
 
     bq query --use_legacy_sql=false "$sql"
 }
+
+slack_top_10_failures() {
+    local job_name_match="${1:-qa}"
+    local subject="${2:-Top 10 QA E2E Test failures for the last 7 days}"
+    local is_test="${3:-true}"
+
+    local sql
+    # shellcheck disable=SC2016
+    sql='
+SELECT
+  MAX(count) AS Failures,
+  Classname AS Suite,
+  Name AS `Case`
```

Comment:
The previous version used double select to handle multiple test from a single suite. Now it's not necessary if we want to have names

Reply:
```
  -- count of test failures in last week
  SELECT
    COUNT(*) AS count,
    Classname AS Suite,
    Name
  FROM
    `acs-san-stackroxci.ci_metrics.stackrox_tests__extended_view`
  WHERE
    CONTAINS_SUBSTR(ShortName, "qa")
    AND Status = "failed"
    AND NOT IsPullRequest
    AND CONTAINS_SUBSTR(JobName, "master")
    AND DATE(Timestamp) >= DATE_SUB(DATE_TRUNC(CURRENT_DATE(), WEEK(MONDAY)), INTERVAL 1 WEEK)
  GROUP BY
    Name,
    Classname
  ORDER BY
    COUNT(*) DESC
LIMIT
  10;
  ```

| count |                      Suite | Name                                                                                                                                                |
|------:|---------------------------:|-----------------------------------------------------------------------------------------------------------------------------------------------------|
|    15 | ExternalNetworkSourcesTest | Verify two flows co-exist if larger network entity added first                                                                                      |
|    15 | ExternalNetworkSourcesTest | Verify connection to a user created external sources                                                                                                |
|    15 | ExternalNetworkSourcesTest | Verify flow stays to the smallest subnet possible                                                                                                   |
|    15 | ExternalNetworkSourcesTest | Verify flow re-maps to larger subnet when smaller subnet deleted                                                                                    |
|    12 |          ImageScanningTest | Quay registry and scanner supports token and/or robot credentials - quay registry+scanner with token and robot creds                                |
|    12 |          ImageScanningTest | Quay registry and scanner supports token and/or robot credentials - quay registry+scanner with token                                                |
|    12 |               VulnMgmtTest | Verify severities and CVSS sha256:481960439934084fb041431f27cb98b89666e1a0daaeb2078bcbe1209790368c ncurses-base MODERATE_VULNERABILITY_SEVERITY 5.3 |
|    12 |               VulnMgmtTest | Verify severities and CVSS sha256:74ee7a5d7a7172090162b1b5f8022b3b403b9f4ac677d325209c56483452f417 ncurses LOW_VULNERABILITY_SEVERITY 9.8           |
|    12 |           IntegrationsTest | Verify Quay Registry+Scanner integration - default config                                                                                           |
|    10 |                    SACTest | Verify GetSummaryCounts using a token with partial access receives partial results                                                                  |

Reply:
:+1: